dataloader.py

```diff
@@ -58,42 +86,51 @@ def PadSequence(batch):
     It will be passed to DataLoader as the keyword argument of collate_fn.
     '''
     
+    #Get the lengths but don't do anything
+    lengths = [x["lengths"] for x in batch]
+    max_length = max(lengths)
+    lengths = torch.Tensor(lengths)
+    
     #Get the sequences, i.e., features and pad them
     sequences = [x["features"] for x in batch]
-    sequences_padded = torch.nn.utils.rnn.pad_sequence(sequences, batch_first=True)
+    sequences_padded = []
+    for s in sequences:
+        while s.shape[0] < max_length:
+            s = torch.cat((s, s[-1,:].view(1,-1)))
+        sequences_padded.append(s)
     
     #Get the labels and pad them
     labels = [x["labels"] for x in batch]
-    labels_padded = torch.nn.utils.rnn.pad_sequence(labels, batch_first=True)
-    
-    #Get the lengths but don't do anything
-    lengths = [x["lengths"] for x in batch]
-    lengths = torch.Tensor(lengths)
+    labels_padded = []
+    for l in labels:
+        while l.shape[0] < max_length:
+            l = torch.cat((l, l[-1].view(1)))
+        labels_padded.append(l)
```

Comment:
It is a little confusing to reassign a variable name you are iterating over. I would recommend making it clearer by changing the name of what you append to something more informative.

Reply:
A question here. If I added a line like this:
`for l in labels:
    single_label = l
    while single_label.shape[0] < max_length:
    ......`
Is this essentially the same as the original code, since I remember the = operator only makes a shallow copy of the list.

Reply:
Oh, I should have been more specific, my bad. My thinking was something even simpler:
```
    for l in labels:
        while l.shape[0] < max_length:
            single_label = torch.cat((l, l[-1].view(1)))
        labels_padded.append(single_label)
```

I'm guessing `troch.cat` makes a deep copy of the list right?

Reply:
Yes, but here I need to keep padding the last line until it reaches the max length in the batch. I actually just realized `l` should be a PyTorch tensor. So probably something like:
```
    for l in labels:
        single_label = l.clone()
        while single_label.shape[0] < max_length:
            single_label = torch.cat((single_label, l[-1].view(1)))
        labels_padded.append(single_label)
```

Then in the end, `labels_padded` should be a list of length `batch_size`, with each element being a tensor of shape `(max_length, 1)`, where `max_length` is the maximum length of the game in the batch.

Reply:
Yeah, your latest suggestion sounds like the right one to use for now. Good clarification.